### ISSUE TITLE
Cannot share database

### ISSUE THREAD

**Issue by dallyn86** [sqlitebrowser/dbhub.io#175]
I have uploaded a database and I tried to add contributors to my database. I added them and saved it but they cannot see the database. Can y'all help?

**Comment by justinclift** [comment]
Looking now. :smile:

**Comment by justinclift** [comment]
What's your account name? :smile:

**Comment by justinclift** [comment]
Hmmm, one potential workaround (for now) if this takes a while to solve, is you can generate your own client certificate + give that to your contributors.  That'll let them remotely log in as you (!), which will mean they can access/write/update the database themselves.

Obviously not a fantastic long term solution, but "it works" is sometimes good enough. :wink:

**Comment by dallyn86** [comment]
My username is dallyn86

**Comment by pedro-lp** [comment]
> Hmmm, one potential workaround (for now) if this takes a while to solve, is you can generate your own client certificate + give that to your contributors. That'll let them remotely log in as you (!), which will mean they can access/write/update the database themselves.
> 
> Obviously not a fantastic long term solution, but "it works" is sometimes good enough. 😉



Hello, I would like you to help me create my database, and be able to integrate it into my sqlite browser application, or if you can send me the documentation link to be able to integrate it, sometimes I use more than one device and I would like to be able to edit my project from my laptop and also from my desktop computer.

**Comment by justinclift** [comment]
Oh, this issue somehow got missed. :frowning:

@dallyn86 If you're still interested in getting sharing working, then it might be a lot easier now.  We updated the web interface a few weeks ago, so shared databases are displayed directly on a person's profile page when they log in.  And it *should* in theory be easier to get sharing working too.

@pedro-lp Can you tell us a bit more about your "SQLite Browser application"? :smile:

Are you meaning you have this application (https://sqlitebrowser.org) running on both your laptop and desktop, and you want to share a (SQLite) database between them both?

*If* that is what you're meaning, then we can probably help with that. :smile:

**Comment by pedro-lp** [comment]
> Oh, this issue somehow got missed. 😦
> 
> @dallyn86 If you're still interested in getting sharing working, then it might be a lot easier now. We updated the web interface a few weeks ago, so shared databases are displayed directly on a person's profile page when they log in. And it _should_ in theory be easier to get sharing working too.
> 
> @pedro-lp Can you tell us a bit more about your "SQLite Browser application"? 😄
> 
> Are you meaning you have this application (https://sqlitebrowser.org) running on both your laptop and desktop, and you want to share a (SQLite) database between them both?
> 
> _If_ that is what you're meaning, then we can probably help with that. 😄

Yes, that's what I mean, how can I edit my database from 2 different devices with the DB Browser for SQLite app? https://sqlitebrowser.org/ if there is any tutorial to do it or documentation, I would appreciate it. 😄

**Comment by justinclift** [comment]
> if there is any tutorial to do it or documentation

I wish there was.  Actually, maybe @chrisjlocke would be interested in whipping up a basic wiki page with some screenshots or similar?

That being said, try this (created just now). :smile:

## The steps to follow

### 1. Create an account on DBHub.io, and log into it
### 2. Generate a client certificate

It's the blue `Generate client certificate` button underneath the `Your page` heading:

![image](https://user-images.githubusercontent.com/406299/225129214-b0eb5a48-2d95-47e7-be4d-0ff2ab2d49e0.png)

That will give you a "certificate" file to save on your local computer.  That file is your ID for DBHub.io, used in the next step. :smile:

### 3. Load the certificate in DB Browser for SQLite

a) Start DB Browser for SQLite
![image](https://user-images.githubusercontent.com/406299/225129923-06f57eb7-3472-4fbb-93b4-17c464097240.png)

b) Open the Preferences dialog
![image](https://user-images.githubusercontent.com/406299/225130088-1e5fcbd5-5816-4a2c-b7bd-4a877a4d5b21.png)

c) Switch to the Remote tab in the Preferences dialog
![image](https://user-images.githubusercontent.com/406299/225130203-72e7cffd-fa49-43f6-ac95-00c985793db3.png)

In my screenshot above, it shows a "justinclift@d..." entry.  Yours will probably be blank instead, as you haven't added a certificate yet.

d) Click the Import certificate file button

It's the top icon on the right side, with a green + sign in the corner of the icon.  (kind of hard to describe it)

That should open a file browser.  Locate that certificate file you downloaded above, and tell it to open that.

The certificate details should then be displayed in the `Your certificates` part of the `Remote` tab, like mine was.

e) Click the "Save" button at the bottom of the Preferences dialog to close it.

At this point, you now have things set up to access DBHub.io. :smile:

### 4. Use the `Remote` tab in DB Browser for SQLite

If it's not displayed yet, then turn on the `Remote` tab.  `View` -> `Remote` from the top menu.

![image](https://user-images.githubusercontent.com/406299/225131538-dfe5bcf5-18cd-4abe-b400-ed3a2c13bdfa.png)

Notice the `Select an identity to connect` drop down?  Click that, and it should show the certificate you just loaded, and also an entry for `Public`.

If you click on the entry for the certificate you just loaded, it will connect to DBHub.io and show your databases on there.  It'll be a blank list, as you've just started out.

When you've gotten this done I'm pretty sure you'll figure the rest out.

* Open a database that's on your computer (in DB Browser for SQLite)
* "Upload" that database using the Remote tab (there's a button for it)
* The database should now show up in the Remote tab, and in the DBHub.io web interface when you're logged in.  And you should be able to access it from any copy of DB Browser for SQLite that you've loaded your certificate file into.  eg laptop and desktop computer

Anyway, try things out and see how you go. :smile:

**Comment by chrisjlocke** [comment]
I'll create a tutorial video - always handy to link to.

**Comment by justinclift** [comment]
@chrisjlocke Good thinking. :smile:

**Comment by chrisjlocke** [comment]
I know.  I surprise myself sometimes. ;)

**Comment by pedro-lp** [comment]
> ¿Ves el menú `Select an identity to connect`desplegable? Haga clic en eso y debería mostrar el certificado que acaba de cargar, y también una entrada para `Public`.
> 
> Si hace clic en la entrada del certificado que acaba de cargar, se conectará a DBHub.io y mostrará sus bases de datos allí. Será una lista en blanco, ya que acabas de empezar.

I followed the steps, but it shows me this message, I don't know if I did something wrong, I downloaded the certificate from the blue button and I only have a private remote database.
"Error opening remote file at https://db4s.dbhub.io/.
The root of the CA certificate is not trusted for this purpose"

<img width="673" alt="image" src="https://user-images.githubusercontent.com/70067671/225157724-6c78f014-0082-46fe-a50b-3f79e5ceab59.png">

**Comment by justinclift** [comment]
Ugh.  That's right, I totally forgot about that.

That's a known issue on macOS (caused by Apple), which has stopped certificates from working on. :frowning: 

https://github.com/sqlitebrowser/sqlitebrowser/issues/2829

We haven't yet found a solution to that problem, although we have some ideas.  It's not going to be fixed in the next few days.

---

That being said, your next option is to just use the DBHub.io web interface instead.  eg upload to your account via the website, and download on both your laptop and desktop using the website too.  And open or edit the files in DB Browser for SQLite afterwards.

It's the same process, just using a web browser for the file transfer instead.

**Comment by pedro-lp** [comment]
> That being said, your next option is to just use the DBHub.io web interface instead. eg upload to your account via the website, and download on both your laptop and desktop using the website too. And open or edit the files in DB Browser for SQLite afterwards.
> 
> It's the same process, just using a web browser for the file transfer instead.

Thank you, I tried it with my laptop, it has windows and it worked correctly, regarding Mac os, yes, it's a problem with the keychain, that's why it doesn't allow to connect correctly
